java/algorithm/src/test/java/org/apache/arrow/algorithm/dictionary/TestSearchDictionaryEncoder.java

```diff
@@ -0,0 +1,345 @@
+         IntVector encodedVector = new IntVector("encoded vector", allocator);
+         VarCharVector dictionary = new VarCharVector("dictionary", allocator)) {
+
+      // set up dictionary
+      dictionary.allocateNew();
+      dictionary.setNull(0);
+      for (int i = 1; i < DICTIONARY_LENGTH; i++) {
+        // encode "i" as i
+        dictionary.setSafe(i, String.valueOf(i).getBytes());
+      }
+      dictionary.setValueCount(DICTIONARY_LENGTH);
+
+      // set up raw vector
+      rawVector.allocateNew(10 * VECTOR_LENGTH, VECTOR_LENGTH);
+      for (int i = 0; i < VECTOR_LENGTH; i++) {
+        if (i % 10 == 0) {
+          rawVector.setNull(i);
+        } else {
+          int val = (random.nextInt() & Integer.MAX_VALUE) % (DICTIONARY_LENGTH - 1) + 1;
+          rawVector.set(i, String.valueOf(val).getBytes());
+        }
+      }
+      rawVector.setValueCount(VECTOR_LENGTH);
+
+      SearchDictionaryEncoder<IntVector, VarCharVector> encoder =
+              new SearchDictionaryEncoder<>(
+                dictionary, DefaultVectorComparators.createDefaultComparator(rawVector), true);
+
+      // perform encoding
+      encodedVector.allocateNew();
+      encoder.encode(rawVector, encodedVector);
+
+      // verify encoding results
+      assertEquals(rawVector.getValueCount(), encodedVector.getValueCount());
+      for (int i = 0; i < VECTOR_LENGTH; i++) {
+        if (i % 10 == 0) {
+          assertEquals(0, encodedVector.get(i));
+        } else {
+          assertArrayEquals(rawVector.get(i), String.valueOf(encodedVector.get(i)).getBytes());
+        }
+      }
+
+      // perform decoding
+      Dictionary dict = new Dictionary(dictionary, new DictionaryEncoding(1L, false, null));
+      try (VarCharVector decodedVector = (VarCharVector) DictionaryEncoder.decode(encodedVector, dict)) {
+
+        // verify decoding results
+        assertEquals(encodedVector.getValueCount(), decodedVector.getValueCount());
+        for (int i = 0; i < VECTOR_LENGTH; i++) {
+          if (i % 10 == 0) {
+            assertTrue(decodedVector.isNull(i));
+          } else {
+            assertArrayEquals(String.valueOf(encodedVector.get(i)).getBytes(), decodedVector.get(i));
+          }
+        }
+      }
+    }
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testEncodeNoNullInDictionary() {
+    try (VarCharVector rawVector = new VarCharVector("original vector", allocator);
+         VarCharVector dictionary = new VarCharVector("dictionary", allocator)) {
+
+      // set up dictionary, with no null in it.
+      dictionary.allocateNew();
+      for (int i = 0; i < DICTIONARY_LENGTH; i++) {
+        // encode "i" as i
+        dictionary.setSafe(i, String.valueOf(i).getBytes());
+      }
+      dictionary.setValueCount(DICTIONARY_LENGTH);
+      
+      // the encoder should encode null, but no null in the dictionary
+      // an exception should be thrown.
+      SearchDictionaryEncoder<IntVector, VarCharVector> encoder =
```

Comment:
I don't think its a problem that the dictionary doesn't have a null in it unless, the vector getting encoded does have a null in it

Reply:
Sounds reasonable. 
The related logic is removed from the constructor, and if there is a null in the vector to encode, an exception will be thrown when encoding.